README.md

```diff
@@ -121,6 +121,56 @@ On your template, provide a button that will allow the user to log in to the app
 </button>
 ```
 
+### Preserve application state through redirects
+
+To preserve application state through the redirect to Auth0 and the subsequent redirect back to your application (if the user authentications succesfully), you can pass in the state that you want preserved to the `loginWithRedirect` method:
+
+```ts
+import { Component } from '@angular/core';
+import { AuthService } from '@auth0/auth0-angular';
+
+@Component({
+  selector: 'app-root',
+  templateUrl: './app.component.html',
+  styleUrls: ['./app.component.css'],
+})
+export class AppComponent {
+  constructor(public auth: AuthService) {}
+
+  loginWithRedirect(): void {
+    this.auth.loginWithRedirect({
+      appState: {
+        // Specify which application state you would like to preserve here (yes, objects ARE allowed)
```

Comment:
I would be a bit more explicit on the sample and drop the emphasis on objects here.

```ts
appState: {
  myValue: 'My State to Track'
}
```